File: Diff5/Problem 41 inc/pandigitalPrime.py
def pandigitalPrime():
    for i in range(1,9000000,2):
        if isPandigital(i) and isPrime(i):
            print i
    
def isPrime(number):
    i = 2
    while i < number / i + 1:
        if number % i == 0:
            return False
        i += 1
    return True

def isPandigital(number):
    reference = ""
    for i in range(1, len(str(number)) + 1):
        reference += str(i)
    for digit in reference:
        reference = reference.replace(digit, "", 1)
        number = str(number).replace(digit,"",1)
    return reference == number 

pandigitalPrime()

File: Diff5/Problem 41 inc/pandigitalPrime.java
public class pandigitalPrime{
    public static void main(String[] commandLine){
        long largest = 0;
        for(long i = 9600001; i > 0; i -= 2) {
            if (isPrime(i) && isPandigital(i)) {
                largest = i;
                break;
            }
        }
        System.out.println(largest);
    }
    public static boolean isPrime(long number){
        for(long i = 2; i < number / i + 1; i++)
            if (number % i == 0) return false;
        return true;
    }
    public static boolean isPandigital(long number) {
        String ref = new String();
        String num = Long.toString(number);
        if (num.indexOf("0") != -1) return false;
        for(int digit = 1; digit <= Long.toString(number).length(); digit++) {
            ref += Integer.toString(digit);
            ref = ref.replaceFirst(Integer.toString(digit), "");
            num = num.replaceFirst(Integer.toString(digit), "");
        }
        return ref.equals(num);   
    }
}